engine/src/test/java/io/camunda/zeebe/engine/state/appliers/EventAppliersTest.java

```diff
@@ -136,4 +141,33 @@ void shouldGetLatestVersionWhenMultipleRegisteredEventAppliersWithDifferentInten
     // then
     Assertions.assertEquals(expectedVersion, actualVersion);
   }
+
+  @Test
+  void shouldRegisterApplierForAllIntents() {
+    // given
+    final var events =
+        Intent.INTENT_CLASSES.stream()
+            .flatMap(c -> Arrays.stream(c.getEnumConstants()))
+            // Heuristic to detect event intents which are generally in past or present tense
+            // instead of the imperative used for commands.
+            .filter(intent -> intent.name().endsWith("ED") || intent.name().endsWith("ING"))
```

Comment:
🔧 Alternatively, we could hardcode this for each intent. If so, we'd need to consider that intents used for commands can also be used for command rejections. Perhaps something like `isCommanding()` would be fitting on the Intent interface. What do you think?

Reply:
> If so, we'd need to consider that intents used for commands can also be used for command rejections

I understand that commands show up in command rejections but I don't quite get how that's relevant here. Could you elaborate?

I like the idea of a method in the `Intent` interface that takes an intent instance and tells you whether it's a command or event :+1: I only though of adding that to the enum members which I discarded because it seemed wasteful to have that data available at runtime. A method solves that nicely though!

Reply:
>I understand that commands show up in command rejections but I don't quite get how that's relevant here. Could you elaborate?
>[..] takes an intent instance and tells you whether it's a command or event

We have three record types. An intent is used either for a:
- command
- command rejection
- event

So an intent can't be just a command or an event. However, intents used for commands can also be used for command rejections. That's why I was hinting at a different wording, something like `isCommanding` or so.

Reply:
Thanks, I've added a method `isEvent` and use that for tests. To ensure that the implementations of that are correct, I've added another test to ensure that only event intents have a registered event applier. Combined with the other test that all event intents have a registered event applier, we also know that the implementations are exhaustive.